security/security.c

```diff
@@ -696,27 +777,29 @@ static void __init lsm_early_task(struct task_struct *task)
  * call_int_hook:
  *	This is a hook that returns a value.
  */
+#define CALL_STATIC_SLOT_VOID(NUM, HOOK, ...)				\
+	static_call_cond(STATIC_SLOT(HOOK, NUM))(__VA_ARGS__);
 
-#define call_void_hook(FUNC, ...)				\
-	do {							\
-		struct security_hook_list *P;			\
-								\
-		hlist_for_each_entry(P, &security_hook_heads.FUNC, list) \
-			P->hook.FUNC(__VA_ARGS__);		\
+#define call_void_hook(FUNC, ...)					\
+	do {								\
+		SECURITY_FOREACH_STATIC_SLOT(CALL_STATIC_SLOT_VOID, 	\
```

Comment:
Maybe clearer to get rid of CALL_STATIC_SLOT_VOID and just put the static_call_cond in directly.

Reply:
`SECURITY_FOREACH_STATIC_SLOT` expects a macro as first argument, so we need to define one even if it's short. Maybe a modification to how `SECURITY_FOREACH_STATIC_SLOT` works could help, but I don't see it.